### ISSUE TITLE
Request textDocument/semanticTokens failed (on edge case)

### ISSUE THREAD

**Issue by scchow** [microsoft/vscode-docker#3836]
# Problem Description

I am encountering the same issue as #2055, which was reported to be fixed a while ago, but has popped up for me in a particular edge case.

This seems to occur when I first end a `RUN` command with `&\`, then add a comment with a markdown link of form `[txt](url)`, and continue the typing below. See the example below for details.

When opening/editing a Dockerfile with this bug, I get repeated notifications that `Request textDocument/semanticTokens/full failed.` Examining the full output, I see:
``` 
[Error - 6:18:50 PM] Request textDocument/semanticTokens/full failed.
  Message: Request textDocument/semanticTokens/full failed with message: Maximum call stack size exceeded
  Code: -32603 
...
<Omitted for brevity>
...
[Error - 6:18:51 PM] Request textDocument/semanticTokens/full failed.
  Message: Request textDocument/semanticTokens/full failed with message: Maximum call stack size exceeded
  Code: -32603 
```
It seems that with every keypress, I get a new message. 

# Sample Dockerfile
``` Dockerfile
FROM ubuntu 

RUN echo "hi there" &\
# Let's add a comment with a markdown link [here](https://www.google.com) hope nothing breaks...
echo "oops I broke something and I'm getting a lot of popup messages"
```

This is a very particular edge case as it only appears if the following conditions are met:
1. End a line with `&\` exactly. Note if you end it with `& \`, this bug does not occur
2. Add a comment with a standard markdown link of style `[txt](url)`
3. Continue the command below the comment

# Setup Details
I was able to reproduce this bug on two setups.
1. Just my primary machine
2. Using Remote-SSH to ssh into a remote machine and editing a Dockerfile on that machine.

## Primary Machine
- OS: Microsoft Windows 10
- VS Code Version: 1.75.1
- Docker Extension Version: v1.23.3
- Remote-SSH Version: v0.96.0

## Remote Machine
- OS: Ubuntu 22.04
- Docker Extension Version: v1.23.3

**Comment by dbreshears** [comment]
@rcjsuen.  Can you take a look at this one?

**Comment by rcjsuen** [comment]
Thank you for the bug report. I have opened https://github.com/rcjsuen/dockerfile-language-service/issues/110 to track this.

**Comment by alexyaang** [comment]
The fix has now been released in Docker extension version 1.26.0
